cmd/tracee/cmd/help_flags.go

```diff
@@ -0,0 +1,36 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/aquasecurity/tracee/pkg/cmd/flags"
+)
+
+func init() {
+	rootCmd.AddCommand(helpFlagsCmd)
+}
+
+var helpFlagsCmd = &cobra.Command{
```

Comment:
This is currently not working for me with `tracee help filter`, or any of the options listed in `flags/help.go`.

Reply:
Ok. `tracee help` only work for commands. Cobra has it as a built-in command help.

Running it we can see:

```
... 
Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  help-flags  Help about any flags
  list        List traceable events
  version     Print the version number of Tracee
...
```

So it works for list as list is a command:

```
./dist/tracee help list
List traceable events

Usage:
  tracee list

Aliases:
  list, l

Flags:
  -h, --help   help for list
```

Cobra distinguishes between command, sub-commands and flags. `--filter` in our case is a flag, so I created the help-flags to output help for them.

Then help is for commands and help-flags is for the flags.

`tracee help`

```
...
Flags:
      --blob-perf-buffer-size int   Size, in pages, of the internal perf ring buffer used to send blobs from the kernel (default 1024)
  -a, --cache strings               Control event caching queues (default [none])
  -C, --capabilities strings        Define capabilities for tracee to run with
  -c, --capture strings             Capture artifacts that were written, executed or found to be suspicious
      --config string               Global config file (yaml)
      --containers                  Enable container info enrichment to events. This feature is experimental and may cause unexpected behavior in the pipeline
      --crs strings                 Define connected container runtimes
  -f, --filter strings              Select events to trace by defining filter expressions
      --healthz                     Enable healthz endpoint
  -h, --help                        help for tracee
      --install-path string         Path where tracee will install or lookup it's resources (default "/tmp/tracee")
      --listen-addr string          Listening address of the metrics endpoint server (default ":3366")
  -l, --log strings                 Logger options (default [info])
      --metrics                     Enable metrics endpoint
  -o, --output strings              Control how and where output is printed (default [table])
  -b, --perf-buffer-size int        Size, in pages, of the internal perf ring buffer used to submit events from the kernel (default 1024)
  -p, --policy strings              Path to a policy or directory with policies
      --pprof                       Enable pprof endpoints
      --pyroscope                   Enable pyroscope agent
      --rego strings                Control event rego settings
      --signatures-dir string       Directory where to search for signatures in CEL (.yaml), OPA (.rego), and Go plugin (.so) formats
```

And `tracee help-flags log` (log is a flag) would do the magic:

```
❯ ./dist/tracee help-flags log   
Control logger options - aggregation and level priority.

Possible options:
  --log aggregate[:interval]          | turns log aggregation on, delaying output optional interval (s, m) (default is off)
  --log <debug|info|warn|error|panic> | set log level, info is the default
  --log file:/path/to/file            | write the logs to a specified file. create/trim the file if exists (default: stderr)

Examples:
  --log debug                              | outputs debug level logs
  --log debug --log aggregate              | outputs aggregated debug level logs every 3 seconds (default)
  --log aggregate:5s                       | outputs aggregated logs every 5 seconds
  --log debug --log file:/tmp/tracee.log   | outputs debug level logs to /tmp/tracee.log
```

Reply:
> Run tracee with a config file (it was ignored when set through the flag)

A flag override the respective value set in the config file.

If a config file has log set to debug and I run `sudo ./dist/tracee --config cfg.yaml --log warn`, all cfg.yaml flags are set but log, which is set to warn.

Reply:
Ok, so what failed for me in the config was the filter field which I set as such:
```
filter:
  - e=ptrace
  - comm=strace
```
Perhaps I set it wrong.

About `help-flags`, i've missed that, thanks for pointing out. Is it impossible to extend the help command for this? It would be more intuitive for use.
Furthermore trying the "old school way" goes like this right now
```
sudo ./dist/tracee --filter help
{"level":"fatal","ts":1681996946.1647704,"msg":"Execution","error":"invalid filter option specified (help), use '--filter help' for more info"}
```

Reply:
> Ok, so what failed for me in the config was the filter field which I set as such:
> 
> ```
> filter:
>   - e=ptrace
>   - comm=strace
> ```
> 
> Perhaps I set it wrong.

It happens that filter, policy and capture cobra flags aren't bind to viper config file. It was a design option. For example, policy has its own logic for file loading and filter use must reside only via CLI.
 
> About `help-flags`, i've missed that, thanks for pointing out. Is it impossible to extend the help command for this? It would be more intuitive for use.

Initially I haven't succeeded in expand the built-in help command to accommodate both command and flags (possibly flag and command conflict or premature parsing of an arg with `--`), there's the reason for a new command to parse only flags without `--`. 

> Furthermore trying the "old school way" goes like this right now
> 
> ```
> sudo ./dist/tracee --filter help
> {"level":"fatal","ts":1681996946.1647704,"msg":"Execution","error":"invalid filter option specified (help), use '--filter help' for more info"}
> ```

Thanks for pointing out this. I'm going to fix the messages for the new binary.

Reply:
> About help-flags, i've missed that, thanks for pointing out. Is it impossible to extend the help command for this? It would be more intuitive for use.
Furthermore trying the "old school way" goes like this right now

Check https://github.com/aquasecurity/tracee/pull/3000#discussion_r1182805909